drivers/serial/Kconfig.emul

```diff
@@ -0,0 +1,19 @@
+# Emulated UART configuration options
+
+# Copyright (c) 2023 Fabian Blatz
+# SPDX-License-Identifier: Apache-2.0
+
+config UART_EMUL
+	bool "Emulated UART driver"
+	default y
+	depends on DT_HAS_ZEPHYR_UART_EMUL_ENABLED
+	select RING_BUFFER
+	help
+	  Enable the emulated UART driver.
+
+config UART_EMUL_BUFFER_SIZE
```

Comment:
Move this to be a `dts` property to allow different sizes for different instances.

Reply:
Done, took some inspiration from other drivers. Hopefully correct :^)